sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/xml/XmlExpressionEvalUtils.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.catalyst.expressions.xml
+
+import org.apache.spark.sql.catalyst.util.GenericArrayData
+import org.apache.spark.sql.types._
+import org.apache.spark.unsafe.types.UTF8String
+
+object XPathEvaluatorFactory {
+  def create(dataType: DataType, path: UTF8String): XPathEvaluator = {
+    dataType match {
+      case BooleanType => XPathBooleanEvaluator(path)
+      case ShortType => XPathShortEvaluator(path)
+      case IntegerType => XPathIntEvaluator(path)
+      case LongType => XPathLongEvaluator(path)
+      case FloatType => XPathFloatEvaluator(path)
+      case DoubleType => XPathDoubleEvaluator(path)
+      case dt if dt.isInstanceOf[StringType] => XPathStringEvaluator(path)
```

Comment:
Because we need to support `Collation` (eg: StringType(...)), we have to write it this way.

Reply:
@panbingkun please add expression-level tests for `xpath_*`

Reply:
> Will it work after your changes? Just in case, are there any tests? cc @stefankandic @uros-db

What I described may have caused `misunderstandings`,
Let me rephrase: This is written so that the case `CollationExpressionWalkerSuite#SPARK-48280: Expression Walker for SQL query examples` can be passed:
https://github.com/apache/spark/blob/be18b7cef3b3162d63930837da6c14469b55759c/sql/core/src/test/scala/org/apache/spark/sql/CollationExpressionWalkerSuite.scala#L735-L745

Reply:
> @panbingkun please add expression-level tests (using `checkEvaluation`) for `xpath_*` functions see for example: `CollationStringExpressionsSuite`

Currently, it is `not supported`. `My description` caused a `misunderstanding`, and I have corrected it.